src/renderer/src/components/Sidebar/SettingsSidebarLink.tsx

```diff
@@ -0,0 +1,32 @@
+import { cloneElement, useEffect, useState } from 'react'
+
+export const SettingsSidebarLink = ({ id, icon, title, onClick, selectedId }): JSX.Element => {
+  const [isActive, setIsActive] = useState<boolean>(id === selectedId)
+
+  const handleClick = () => {
+    setIsActive(!isActive)
+    onClick?.()
+  }
+  useEffect(() => {
+    setIsActive(id === selectedId)
+  }, [selectedId, id])
+
+  return (
+    <li className="my-3">
+      <button
+        onClick={() => handleClick()}
```

Comment:
If you follow the instructions above, you can change it to:
```suggestion
        onClick={onClick}
```